### PR TITLE
Upgrade clickhouse from v22.8.9.24 to v23.8.7.24

### PR DESCRIPTION
**Pull request by citrus-it**
The tests all now pass and I've deployed this to a bench gimlet and done some basic smoke tests.

- [x] Single node deployment;
- [x] Clustered deployment;
- [x] Upgrade from old version to new;
- [x] More tests, TBD

### CODE REVIEW

File: oximeter/db/src/client.rs [hunk 1]
```diff
@@ -857,9 +857,11 @@ impl Client {
             Err(Error::Database(err))
                 // Case 1: The database has not been created.
                 if err.contains("Database oximeter doesn't exist") ||
+                    err.contains("Database oximeter does not exist") ||
```
Comment (bnaecker) [inline oximeter/db/src/client.rs:860]: Nit, but maybe extract this to a const?

File: oximeter/db/src/configs/keeper_config.xml [hunk 2]
```diff
@@ -9,13 +9,14 @@
...
     <!-- To allow all use :: -->
-    <listen_host from_env="CH_LISTEN_ADDR"/> 
+    <listen_host from_env="CH_LISTEN_ADDR"/>
     <path replace="true" from_env="CH_DATASTORE"/>
 
     <keeper_server>
         <tcp_port from_env="CH_LISTEN_PORT"/>
         <server_id from_env="CH_KEEPER_ID_CURRENT"/>
         <log_storage_path from_env="CH_LOG_STORAGE_PATH"/>
+        <storage_path from_env="CH_STORAGE_PATH"/>
```
Comment (bnaecker) [inline oximeter/db/src/configs/keeper_config.xml:19]: I can't find any documentation for this new setting, have you? What does it refer to?

**Review by bnaecker — APPROVED** [review]
Looks good to me, thanks for pushing on this!

File: oximeter/db/src/configs/keeper_config.xml (hunk 2 above)
Comment (citrus-it) [reply oximeter/db/src/configs/keeper_config.xml]: I meant to remove that, it was a red herring when I was troubleshooting earlier.
If it isn't explicitly set, it just gets set to the parent directory of `<log_storage_path/>` - it seems to be a way of just defining a parent directory for all of the other pieces, but doesn't seem to be documented.
I expect the tests will pass without it.

**Review by karencfv — APPROVED** [review]
Thanks for looking into this, and also for cleaning up the tests!

File: oximeter/db/src/client.rs (hunk 1 above)
Comment (citrus-it) [reply oximeter/db/src/client.rs]: Done

**Pull request merged**